### DISCUSSION TITLE
Givetalentinfo can't be used to give talents that could be unlocked through normal means

### DISCUSSION THREAD

**Discussion by TheNinjaScout** [FakeFishGames/Barotrauma#10873] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Givetalentinfo has a parameter called "giverandom" which will cause the status effect to give a random talent out of all the defined "talentidentifiers" to the target character.

If "giverandom" is set to false, it will instead apply all the talents defined in the "talentidentifiers" section.

When giverandom is set to true, it will not allow the character to be given any of the talents that the character would have been able to unlock from their available talent trees, as to ensure they can be given something somewhat interesting, and this works as intended, however when giverandom is set to false, the status effect will still not be able to give talents that are unlockable in the character's skill tree, which seems to contradict the comment left [here](https://github.com/Regalis11/Barotrauma/blob/a4cee74f8bdf9ed52f639ff3950ff507155ff5cd/Barotrauma/BarotraumaShared/SharedSource/StatusEffects/StatusEffect.cs#L1601 ).

To fix the issue, [this](https://github.com/Regalis11/Barotrauma/blob/a4cee74f8bdf9ed52f639ff3950ff507155ff5cd/Barotrauma/BarotraumaShared/SharedSource/StatusEffects/StatusEffect.cs#L1610) line of code should be changed to this:

```c#
if (targetCharacter.Info.UnlockedTalents.Contains(id)) { continue; }
```

Which, when "giverandom" is set to false, will only prevent the talent from being given if the character already has it.

### Bug prevalence

Happens every time I play

### Version

0.20.16.1

### Which operating system did you encounter this bug on?

Windows

**Comment by 3e849f2e5c** [comment]
The reason why you can't give talents that are unlockable in the talent trees is that it would completely break the UI, if you would for example get a T5 talent while you still haven't finished your primary talents it would lose count of the unlocked talents and also cause talent trees to be locked out without your own input since you would have to finish that one tree before starting a new one now. I don't see a feasible implementation where this would play nicely to be honest.

**Comment by Regalis11** [comment]
Took a quick look at this, and it does seem like changing this would require significant changes to the talent tree logic as @3e849f2e5c said. I'd also consider this a feature request rather than a bug report, so I'll convert this to a discussion.
